### PR TITLE
Update editorconfig naming rules

### PR DESCRIPTION
**Pull request by russcam**
Relates: #1288

/cc @stevejgordon

### CODE REVIEW

**Comment by russcam** [comment]
@stevejgordon Would you mind taking a look at these rules? They're largely insipired by [Roslyn's editorconfig rules](https://github.com/dotnet/roslyn/blob/3fa5dff8c1ffbfab5b8f6a73598cf5e3cca4ec6d/.editorconfig) and specifically address a few things like:

- Using Pascal case for constants
- Ensuring all local functions are Pascal case
- Locals and parameters are camel case

File: .editorconfig [hunk 1]
```diff
@@ -147,3 +211,6 @@ insert_final_newline=true
 
 [*.sh]
 end_of_line = lf
+
+[{LICENSE,NOTICE}]
```
Comment (Mpdreamz) [inline .editorconfig:215]: :+1:

**Review by Mpdreamz — APPROVED** [review]
LGTM

**Review by gregkalapos — APPROVED** [review]

File: .editorconfig (hunk 1 above)
Comment (russcam) [reply .editorconfig]: split out into separate PR

**Pull request merged**